src/main/java/teammates/common/datatransfer/FeedbackRubricQuestionDetails.java

```diff
@@ -66,7 +75,33 @@ public boolean extractQuestionDetails(
         
         return true;
     }
-    
+
+    private List<Double> getRubricWeights(Map<String, String[]> requestParameters, int numOfRubricChoices, boolean hasAssignedWeights) {
+        List<Double> rubricWeights = new ArrayList<Double>();
+
+        if (!hasAssignedWeights) {
+            return rubricWeights;
+        }
+
+        for (int i = 0; i < numOfRubricChoices; i++) {
+
+            String weight = HttpRequestHelper.getValueFromParamMap(requestParameters, Const.ParamsNames.FEEDBACK_QUESTION_RUBRIC_WEIGHT + "-" + i);
+            String choice = HttpRequestHelper.getValueFromParamMap(requestParameters, Const.ParamsNames.FEEDBACK_QUESTION_RUBRIC_CHOICE + "-" + i);
+
+            if (choice == null) {
+                continue;
+            }
+
+            try {
+                rubricWeights.add(Double.parseDouble(weight));
+            } catch (NumberFormatException e) {
+                // Do not add weight to rubricWeights if the weight cannot be parsed
```

Comment:
The `type` of the input boxes for the weights is `number`. If the user adds unparsable weights, the browser will prevent form submission and show a warning message. However, it is still possible for the user to assign empty weights, therefore we need to check them in `FeedbackRubricQuestionDetails`.

![error](https://cloud.githubusercontent.com/assets/10228385/14308885/ad612b7c-fc0b-11e5-8b42-70710d684e44.jpg)
The browser that I am using is Chrome.